### PR TITLE
Retry sync on failure

### LINKED ISSUE #147: [Offline support] Retry failed mutation sync attempts
- [x] Respect backoff if successive edits enqueue updates to same feature
- [x] Increment retry counter
- [ ] Report error if retry counter exceeds some threshold

### PR DESCRIPTION
**Pull request by gino-m**
Closes #147. Future work in #154.

For now retry indefinitely.

Could one you kindly give a quick look? Please reply here if you're looking so the other knows. Thanks in advance!

### CODE REVIEW

**Comment by gino-m** [comment]
Note: Contains diffs that will disappear after merging #153.

**Comment by scolsen** [comment]
I'll take a look.

File: gnd/src/main/java/com/google/android/gnd/persistence/shared/FeatureMutation.java [hunk 1]
```diff
@@ -40,6 +40,17 @@
...
   public abstract Optional<Point> getNewLocation();
 
+  @Override
+  public abstract Builder toBuilder();
+
+  /** Returns the mutations of type {@link FeatureMutation} contained in the specified list. */
+  public static ImmutableList<FeatureMutation> filter(ImmutableList<Mutation> mutations) {
+    return stream(mutations)
+        .filter(FeatureMutation.class::isInstance)
+        .map(FeatureMutation.class::cast)
```
Comment (scolsen) [inline gnd/src/main/java/com/google/android/gnd/persistence/shared/FeatureMutation.java]: Why do we need to cast here? Does the compiler require this--is it unable to figure out the type in this case without explicit casting? Wouldn't filter(FeatureMutation.class::isInstance) already concertize to a type: ImmutableList<FeatureMutation>?

Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/persistence/shared/FeatureMutation.java]: We can treat `FeatureMutation` as the more generic type `Mutation`, but here we're down casting the items, which in Java is always explicit.

**Comment by gino-m** [comment]
> Looks good, thanks!

Thanks for reviewing!

**Pull request merged**